pkgs/servers/web-apps/slskd/default.nix

```diff
@@ -0,0 +1,71 @@
+{ lib, stdenv
+, fetchFromGitHub
+, fetchurl
+, unzip
+, dotnetCorePackages
+, buildDotnetModule
+, mono
+}:
+let
+  pname = "slskd";
+  version = "0.17.5";
+
+  # TODO build this via npm
+  wwwroot = stdenv.mkDerivation {
+    inherit version;
+    pname = "${pname}-wwwroot";
+
+    src = fetchurl {
+      url = "https://github.com/slskd/slskd/releases/download/${version}/slskd-${version}-linux-x64.zip";
+      sha256 = "sha256-QYGu2kwZJPMXeSYcE2HpC44RvSOl4MUgNp1Z+gTvrck=";
+    };
+
+    nativeBuildInputs = [ unzip ];
+
+    unpackPhase = ''
+      unzip -q $src
+    '';
+
+    sourceRoot = "./wwwroot";
+
+    installPhase = ''
+      cp -r . $out
+    '';
+  };
+
+in buildDotnetModule {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    owner = "slskd";
+    repo = "slskd";
+    rev = version;
+    sha256 = "sha256-iIM29ZI3M9etbw4yzin+4f4cGHIt5qjIl7uzsTUCBc4=";
+  };
+
+  runtimeDeps = [ mono ];
```

Comment:
Is this required? It builds and runs without this in my tests. I was building against 22.11 though, not master.

Reply:
I can confirm I need mono. Without this, it fails at runtime:
```

                   ▄▄▄▄         ▄▄▄▄       ▄▄▄▄
           ▄▄▄▄▄▄▄ █  █ ▄▄▄▄▄▄▄ █  █▄▄▄ ▄▄▄█  █
           █__ --█ █  █ █__ --█ █    ◄█ █  -  █
           █▄▄▄▄▄█ █▄▄█ █▄▄▄▄▄█ █▄▄█▄▄█ █▄▄▄▄▄█
╒════════════════════════════════════════════════════════╕
│           GNU AFFERO GENERAL PUBLIC LICENSE            │
│                   https://slskd.org                    │
│                                                        │
│                    0.17.5 (0.17.5)                     │
└────────────────────────────────────────────────────────┘
[19:00:58 INF] Version: 0.17.5 (0.17.5)
[19:00:58 INF] System: .NET 7.0.3, Unix 6.1.20.0, 64 bit, 8 processors
[19:00:58 INF] Process ID: 11718 (64 bit)
[19:00:58 INF] Invocation ID: 33158f30-5ece-4a35-83f2-2c755d654b2a
[19:00:58 INF] Instance Name: default
[19:00:58 INF] Using application directory /home/ao/.local/share/slskd
[19:00:58 INF] Using configuration file /home/ao/.local/share/slskd/slskd.yml
[19:00:58 INF] Storing application data in /home/ao/.local/share/slskd/data
[19:00:59 INF] Listening for HTTP requests at http://0.0.0.0:5030/
[19:00:59 INF] Listening for HTTPS requests at https://0.0.0.0:5031/
[19:00:59 INF] Using randomly generated self-signed certificate
[19:00:59 INF] Using base url /
[19:00:59 INF] Serving static content from /nix/store/4hsc0rfh3brs94mqjb95v65165klriyw-slskd-0.17.5/lib/slskd/wwwroot
[19:01:00 FTL] Application terminated unexpectedly
System.DllNotFoundException: Unable to load shared library 'MonoPosixHelper' or one of its dependencies. In order to help diagnose loading problems, consider using a tool like strace. If you're using glibc, consider setting the LD_DEBUG environment variable:
/nix/store/4hsc0rfh3brs94mqjb95v65165klriyw-slskd-0.17.5/lib/slskd/runtimes/linux-x64/native/MonoPosixHelper.so: cannot open shared object file: No such file or directory
/nix/store/fbx2azygdqr455mwxhmdjl4gsgxj2xdz-aspnetcore-runtime-7.0.3/shared/Microsoft.NETCore.App/7.0.3/MonoPosixHelper.so: cannot open shared object file: No such file or directory
/nix/store/4hsc0rfh3brs94mqjb95v65165klriyw-slskd-0.17.5/lib/slskd/runtimes/linux-x64/lib/netstandard2.0/MonoPosixHelper.so: cannot open shared object file: No such file or directory
libz.so.1: cannot open shared object file: No such file or directory
/nix/store/fbx2azygdqr455mwxhmdjl4gsgxj2xdz-aspnetcore-runtime-7.0.3/shared/Microsoft.NETCore.App/7.0.3/libMonoPosixHelper.so: cannot open shared object file: No such file or directory
/nix/store/4hsc0rfh3brs94mqjb95v65165klriyw-slskd-0.17.5/lib/slskd/runtimes/linux-x64/lib/netstandard2.0/libMonoPosixHelper.so: cannot open shared object file: No such file or directory
/nix/store/4hsc0rfh3brs94mqjb95v65165klriyw-slskd-0.17.5/lib/slskd/runtimes/linux-x64/native/MonoPosixHelper: cannot open shared object file: No such file or directory
/nix/store/fbx2azygdqr455mwxhmdjl4gsgxj2xdz-aspnetcore-runtime-7.0.3/shared/Microsoft.NETCore.App/7.0.3/MonoPosixHelper: cannot open shared object file: No such file or directory
/nix/store/4hsc0rfh3brs94mqjb95v65165klriyw-slskd-0.17.5/lib/slskd/runtimes/linux-x64/lib/netstandard2.0/MonoPosixHelper: cannot open shared object file: No such file or directory
/nix/store/4hsc0rfh3brs94mqjb95v65165klriyw-slskd-0.17.5/lib/slskd/runtimes/linux-x64/native/libMonoPosixHelper: cannot open shared object file: No such file or directory
/nix/store/fbx2azygdqr455mwxhmdjl4gsgxj2xdz-aspnetcore-runtime-7.0.3/shared/Microsoft.NETCore.App/7.0.3/libMonoPosixHelper: cannot open shared object file: No such file or directory
/nix/store/4hsc0rfh3brs94mqjb95v65165klriyw-slskd-0.17.5/lib/slskd/runtimes/linux-x64/lib/netstandard2.0/libMonoPosixHelper: cannot open shared object file: No such file or directory

   at Mono.Unix.Native.Syscall.sysconf(SysconfName name, Errno defaultError)
   at Mono.Unix.Native.Syscall.sysconf(SysconfName name)
   at Prometheus.SystemMetrics.Collectors.LinuxCpuUsageCollector.CreateMetrics(MetricFactory factory)
   at Prometheus.SystemMetrics.SystemMetricsHostedService.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.Internal.Host.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.Run(IHost host)
   at slskd.Program.Main(String[] args) in /build/source/src/slskd/Program.cs:line 467
```

I tried changing the `projectFile` which doesn't change anything. Maybe you already have it on your computer and .NET somehow finds it?

I found mono as a runtime dependency in [another package](https://github.com/NixOS/nixpkgs/blob/9b04b1d/pkgs/servers/nosql/eventstore/default.nix#L32).

Reply:
No problem :smile: